tests/components/google/test_calendar.py

```diff
@@ -1359,3 +1361,90 @@ async def test_invalid_rrule_fix(
     assert event["uid"] == "cydrevtfuybguinhomj@google.com"
     assert event["recurrence_id"] == "_c8rinwq863h45qnucyoi43ny8_20230915"
     assert event["rrule"] is None
+
+
+@pytest.mark.parametrize(
+    ("event_type", "expected_event_message"),
+    [
+        ("default", "Test All Day Event"),
+        ("workingLocation", None),
+    ],
+)
+async def test_working_location_ignored(
+    hass: HomeAssistant,
+    hass_client: ClientSessionGenerator,
+    mock_events_list_items: Callable[[list[dict[str, Any]]], None],
+    component_setup: ComponentSetup,
+    event_type: str,
+    expected_event_message: str | None,
+) -> None:
+    """Test working location events are skipped."""
+    event = {
+        **TEST_EVENT,
+        **upcoming(),
+        "eventType": event_type,
+    }
+    mock_events_list_items([event])
+    assert await component_setup()
+
+    state = hass.states.get(TEST_ENTITY)
+    assert state
+    assert state.name == TEST_ENTITY_NAME
+    assert state.attributes.get("message") == expected_event_message
+
+
+@pytest.mark.parametrize("calendar_is_primary", [True])
+async def test_working_location_entity(
+    hass: HomeAssistant,
+    hass_client: ClientSessionGenerator,
+    entity_registry: er.EntityRegistry,
+    mock_events_list_items: Callable[[list[dict[str, Any]]], None],
+    component_setup: ComponentSetup,
+) -> None:
+    """Test that working location events are registered under a disabled by default entity."""
+    event = {
+        **TEST_EVENT,
+        **upcoming(),
+        "eventType": "workingLocation",
+    }
+    mock_events_list_items([event])
+    assert await component_setup()
+
+    entity_entry = entity_registry.async_get("calendar.working_location")
+    assert entity_entry
+    assert entity_entry.disabled_by == RegistryEntryDisabler.INTEGRATION
+
+    entity_registry.async_update_entity(
+        entity_id="calendar.working_location", disabled_by=None
+    )
+    async_fire_time_changed(
+        hass,
+        dt_util.utcnow() + datetime.timedelta(seconds=RELOAD_AFTER_UPDATE_DELAY + 1),
+    )
```

Comment:
Did the freezer work in these tests?

Reply:
I've found it bit difficult for some calendar given everything involved (all the timers to do entity updates, look at calendar state and handle state changes, etc) so these tests don't use it. I think freezer wasn't as popular when these were originally written, but it may be possible to move them all to use freezer, I haven't tried in about a year.

I can take a pass at moving some of these and see how hard it is.

Reply:
Oh it was just to verify, I can remember you tried but it didn't work, otherwise I would've commented it.

But let's move that to a followup then

Reply:
Yeah I didn't try it for this test, i'm copying the pattern of existing tests here.